scripts/sonic_sku_create.py

```diff
@@ -0,0 +1,853 @@
+
+            port_inst = {}
+            j = 1
+            lanes_arr = lanes_str_result.split(':')
+            step = bko_dict[port_split]["step"]
+            alias_arr = ['a','b','c','d']
+	    pattern = '^([0-9]{1,})x([0-9]{1,})'
+	    m = re.match(pattern,port_split)
+            speed = int(m.group(2))
+            bko = bko_dict[port_split]["bko"]
+
+            for i in range(0,self.base_lanes,step):
+                port_str = "Ethernet{:d}".format(port_idx + i/step)
+                lanes_str = lanes_arr[j]
+                j += 1
+                 
+                if bko == 0:
+                    alias_str = "etp{:d}".format((port_idx/self.base_lanes)+1)
+                else:
+                    alias_str = "etp{:d}{:s}".format((port_idx/self.base_lanes)+1,alias_arr[i/step])
+                    print ("i= ",i," alias_str= ",alias_str)
+                port_inst["lanes"] = lanes_str
+                port_inst["alias"] = alias_str
+                port_inst["speed"] = speed*1000
+                port_inst["mtu"] = mtu
+
+                xxx = copy.deepcopy(port_inst)
+                data['PORT'][port_str] = xxx
+                print (port_str, data['PORT'][port_str])
+
+            with open(new_file, 'w') as outfile:
+                json.dump(data, outfile, indent=4, sort_keys=True)
+
+            print ("--------------------------------------------------------")
+
+        def break_a_port(self,port_name,port_split):
+            new_file = self.ini_file + ".new"
+            lanes_str_result = self.break_in_ini(self.ini_file,port_name,port_split)
+            self.port_config_split_analyze(self.ini_file)
+            self.form_port_config_dict_from_ini(self.ini_file)
+	    self.platform_specific()	
+            shutil.copy(new_file,self.ini_file)
+            if lanes_str_result == None:
+                print ("break_in_ini function returned empty lanes string, Exiting...")
+                exit(1)
+            print ("Before break_in_cfg")    
+            self.break_in_cfg(self.cfg_file,port_name,port_split,lanes_str_result)
+
+
+		
+	def split_analyze(self) :
+		# Analyze the front panl ports split  based on the interfaces alias names
+		# fpp_split is a hash with key=front panel port and values is a list of lists ([alias],[index])
+		alias_index = PORTCONFIG_HEADER.index('alias')
+		for idx,ifc in self.portconfig_dict.items():
+			pattern = '^etp([0-9]{1,})([a-d]?)'
+			m = re.match(pattern,str(ifc[alias_index]))
+			if int(m.group(1)) not in self.fpp_split :
+				self.fpp_split[int(m.group(1))] = [[ifc[alias_index]],[idx]] #1
+			else:
+				self.fpp_split[int(m.group(1))][0].append(str(ifc[alias_index])) #+= 1
+				self.fpp_split[int(m.group(1))][1].append(idx)
+			if (self.verbose) :
+				print("split_analyze -> ",m.group(1), " : ", self.fpp_split[int(m.group(1))])
+		self.num_of_fpp = len(self.fpp_split.keys())
+		
+	def get_default_lanes(self) :
+
+		try:
+			f = open(self.base_file_path,"r")
```

Comment:
@madhanmellanox consider using `with` concept

Reply:
Addressed in next commit.